parallaxinc/solo
# Solo 81c move terminal into single function

Final PR to move terminal functionality into its own class.
Addresses #81 

Done using ES5 methodology - which means that because internally defined functions are not elevated, most (if not all) of the functions had to be re-ordered, so this PR will be unfortunately hard-to-read.

I tested this as extensively as I could.  I used both the older and newer (WS and non-WS) clients, two board-types, simulated overruns, character positioning, transmitting and receiving characters, and everything appears to work correctly (or better than before).

**To test**
Run the following program:
![image](https://user-images.githubusercontent.com/19737272/70281144-2652f100-176f-11ea-90c8-4541b4683e29.png)
You should see a prompt, then enter text.  You should see that text repeated back, then a cursor positioning test output.
